lib/reporting/cloudwatch_client.rb

```diff
@@ -123,7 +123,7 @@ def fetch(query:, from: nil, to: nil, time_slices: nil)
       # rubocop:enable Metrics/BlockLength
 
       if each_result_queue
-        threads << Thread.new do
+        result_thread = Thread.new do
```

Comment:
for explanation: adding this result outputting thread to the array of "worker" threads (maybe we should rename `threads` to `worker_threads` for clarity) was causing a bit of a dependency error, since the worker threads submit work that needs to get processed by this output thread